src/entities/removeLiquidity/helper.ts

```diff
@@ -0,0 +1,83 @@
+import {
+    MAX_UINT256,
+    removeLiquiditySingleTokenExactInShouldHaveTokenOutIndexError,
+} from '@/utils';
+import { Token } from '../token';
+import { RemoveLiquidityAmounts } from '../types';
+import { getAmounts } from '../utils';
+import {
+    RemoveLiquidityCall,
+    RemoveLiquidityInput,
+    RemoveLiquidityKind,
+} from './types';
+
+export const getAmountsQuery = (
+    tokens: Token[],
+    input: RemoveLiquidityInput,
+): RemoveLiquidityAmounts => {
+    switch (input.kind) {
+        case RemoveLiquidityKind.Unbalanced:
+            return {
+                minAmountsOut: getAmounts(tokens, input.amountsOut),
+                tokenOutIndex: undefined,
+                maxBptAmountIn: MAX_UINT256, // maxAmountIn set to max when querying
+            };
+        case RemoveLiquidityKind.SingleTokenExactOut:
+            return {
+                minAmountsOut: getAmounts(tokens, [input.amountOut]),
+                tokenOutIndex: tokens.findIndex((t) =>
+                    t.isSameAddress(input.amountOut.address),
+                ),
+                maxBptAmountIn: MAX_UINT256, // maxAmountIn set to max when querying
+            };
+        case RemoveLiquidityKind.SingleTokenExactIn:
+            return {
+                minAmountsOut: Array(tokens.length).fill(1n), // minAmountsOut set to 1 wei when querying
```

Comment:
Is this right?

Reply:
Yeah, that was intentional. But we don't need to worry too much about it.
Giving a bit more context, on v3, SC expects minAmountsOut to be non-zero. 
Specially on the SingleTokenExactOut case, where they expect minAmountsOut to have a single non-zero value, at the index of the tokenOut.
That's how they identify which token to remove liquidity to.
But, having said that, the plan is to remove minAmountsOut from query inputs as a whole, as they don't see value on having it there.
So we'll have to update this when the new Router is released.
Note: on the actual transaction call, they will enforce minAmountsOut to be non-zero